pennylane/labs/tests/resource_estimation/test_resource_config.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test the resource configuration class.
+"""
+import pytest
+
+from pennylane.labs.resource_estimation.ops.op_math.controlled_ops import (
+    ResourceCRX,
+    ResourceCRY,
+    ResourceCRZ,
+)
+from pennylane.labs.resource_estimation.ops.qubit.parametric_ops_single_qubit import (
+    ResourceRX,
+    ResourceRY,
+    ResourceRZ,
+)
+from pennylane.labs.resource_estimation.resource_config import ResourceConfig
+from pennylane.labs.resource_estimation.resource_operator import ResourceOperator
+from pennylane.labs.resource_estimation.templates import ResourceSelectPauliRot
+
+# pylint: disable=protected-access, unused-argument
+
+
+class DummyOp(ResourceOperator):
+    """Dummy ResourceOperator for testing."""
+
+    num_wires = 1
+
+
+def dummy_decomp_func(**kwargs):
+    """Dummy decomposition function for testing."""
+    return []
+
+
+class TestResourceConfig:
+    """Test the ResourceConfig class and its methods."""
+
+    def test_initialization(self):
+        """Test that the ResourceConfig class initializes correctly with default values."""
+        config = ResourceConfig()
+
+        assert isinstance(config.errors_and_precisions, dict)
+        assert config.errors_and_precisions[ResourceRX] == {"precision": 1e-9}
+        assert config.errors_and_precisions[ResourceCRZ] == {"precision": 1e-9}
+
+        assert not config._custom_decomps
+        assert not config._adj_custom_decomps
+        assert not config._ctrl_custom_decomps
+        assert not config._pow_custom_decomps
```

Comment:
This is one of those things pylint complains about but we try and go against.
```suggestion
        assert config._pow_custom_decomps == {}
```
It is more clear what type we expect.